src/components/SearchInput/index.js

```diff
@@ -18,49 +18,92 @@ class SearchInput extends Component {
     super(props);
 
     this.state = {
-      isFocused: !props.hasBackground
+      isFocused: false
     };
 
     this.inputRef = React.createRef();
-    this.inputBlur = this.inputBlur.bind(this);
-    this.inputFocused = this.inputFocused.bind(this);
+    this.inputClicked = false;
   }
 
-  inputBlur() {
-    const { hasBackground, onBlur } = this.props;
+  componentDidMount() {
+    global.window.addEventListener("click", this.windowClick);
```

Comment:
Please only add the event listener when the input is focused and remove it when the input is blurred.